File: Nvisibl.DataLibrary/Models/Message.cs
using System;
using System.ComponentModel.DataAnnotations;

namespace Nvisibl.DataLibrary.Models
{
    public class Message
    {
        public int Id { get; set; }

        [Required]
        [MaxLength(4000)]
        public string Body { get; set; } = string.Empty;

        [Required]
        public DateTime TimeSentUtc { get; set; }

        [Required]
        public int AuthorId { get; set; }

        [Required]
        public int ChatroomId { get; set; }

        public User? Author { get; set; }

        public Chatroom? Chatroom { get; set; }
    }
}

File: Nvisibl.WebUI/src/models/message.ts
export default class Message {
    constructor(
        id: number,
        authorId: number,
        chatroomId: number,
        body: string,
        timeSentUTC: Date,
    ) {
        this.id = id;
        this.authorId = authorId;
        this.chatroomId = chatroomId;
        this.body = body;
        this.timeSentUTC = timeSentUTC;
    }

    public readonly id: number;
    public readonly authorId: number;
    public readonly chatroomId: number;
    public readonly body: string;
    public readonly timeSentUTC: Date;
}
